sdk/core/azure-core/inc/azure/core/http/websockets/websockets_transport.hpp

```diff
@@ -0,0 +1,181 @@
+       * to the remote node or the final frame of a multipart message.
+       */
+      FrameTypeBinary,
+      /**
+       * @brief Indicates that the frame is a partial binary frame - it is NOT the
+       * complete frame to be sent to the remote node.
+       */
+      FrameTypeBinaryFragment,
+
+      FrameTypeClosed,
+    };
+    /**
+     * @brief Destructs `%HttpTransport`.
+     *
+     */
+    virtual ~WebSocketTransport() {}
+
+    /**
+     * @brief Determines if the transport natively supports WebSockets or not.
+     *
+     * @returns true iff the transport has native websocket support, false otherwise.
+     */
+    virtual bool NativeWebsocketSupport() = 0;
+    /**
+     * @brief Complete the WebSocket upgrade.
+     *
+     * @details Called by the WebSocket client after the HTTP server responds with a
+     * SwitchingProtocols response. This method performs whatever operations are needed to
+     * transfer the protocol from HTTP to WebSockets.
+     */
+    virtual void CompleteUpgrade() = 0;
+
+    /**************/
+    /* Native WebSocket support functions*/
+    /**************/
+    /**
+     * @brief Gracefully closes the WebSocket, notifying the remote node of the close reason.
+     *
+     * @param status Status value to be sent to the remote node. Application defined.
+     * @param disconnectReason UTF-8 encoded reason for the disconnection. Optional.
+     * @param context Context for the operation.
+     */
+    virtual void CloseSocket(
+        uint16_t status,
+        std::string const& disconnectReason,
+        Azure::Core::Context const& context)
+        = 0;
+
+    /**
+     * @brief Closes the WebSocket.
+     *
+     * Does not notify the remote endpoint that the socket is being closed.
+     *
+     */
+    virtual void Close() = 0;
+
+    /**
+     * @brief Retrieve the information associated with a WebSocket close response.
+     *
+     * @param context Context for the operation.
+     *
+     * @returns a tuple containing the status code and string.
+     */
+    virtual std::pair<uint16_t, std::string> GetCloseSocketInformation(
+        Azure::Core::Context const& context)
+        = 0;
+
+    /**
+     * @brief Send a frame of data to the remote node.
+     *
+     * @param frameType Frame type sent to the server, Text or Binary.
+     * @param frameData Frame data to be sent to the server.
+     * @param context Context for the operation.
+     */
+    virtual void SendFrame(
+        WebSocketFrameType frameType,
+        std::vector<uint8_t> const& frameData,
+        Azure::Core::Context const& context)
+        = 0;
+
+    /**
+     * @brief Receive a frame from the remote WebSocket server.
+     *
+     * @param frameTypeReceived frame type received from the remote server.
+     *
+     * @returns Frame data received from the remote server.
+     */
+    virtual std::vector<uint8_t> ReceiveFrame(
+        WebSocketFrameType& frameTypeReceived,
```

Comment:
is this an output parameter? what happened to 

std::shared_ptr<WebSocketFrame> ReceiveFrame(
        Azure::Core::Context const& context = Azure::Core::Context{});